docs/en/02_Developer_Guides/14_Files/05_File_Migration.md

```diff
@@ -114,6 +114,121 @@ Use the following estimates to decide how you will run your file migration:
+
+Silverstripe 4.0 to 4.3 were using hash paths for the public file store as well as the protected file store. This meant that every time a new version of a file was published, it would be served under a different URL.
+
+SilverStripe 4.4 and later default to using natural paths for public store files. After upgrading to SilverStripe 4.4 or above, you can run the file migration task to normalise your existing files.
+
+This will take all existing files under a hash path and move them to their natural path.
+
+### Do I need to re-migrate my files?
+
+No, you do not need to run the file migration task when upgrading to SilverStripe 4.4 or later. 
+
+Your existing files will still resolve. Old hash paths will be redirected to the latest version of the file. As newer versions of existing files are published, they will be moved to their natural path.
+
+You can choose to run the migration task at any point.
+
+### What if I’m using `legacy_filenames`?
+
+SilverStripe 4.0 to 4.3 allowed you to enable a `legacy_filenames` mode. This would be equivalent to using natural path for both the public store and the protected store.
+
+With the introduction of natural path for the public store, the primary reason for `legacy_filenames` mode is no longer relevant. `legacy_filenames` mode is now considered deprecated, however leaving it enabled will not have any impact on your project – it will simply be ignored.
+
+After upgrading to SilverStripe 4.4 or later, you should run the file migration task to normalise your protected files. This will move your protected files to their equivalent hash path.
+
+###  What if I want to keep using hash paths for my public files?
+
+If you want to keep using hash paths for your public store, you can add the following entries to your YML configuration files.
+
+```yml
+SilverStripe\Core\Injector\Injector:
+  # Define public resolution strategy
+  SilverStripe\Assets\FilenameParsing\FileResolutionStrategy.public:
+    class: SilverStripe\Assets\FilenameParsing\FileIDHelperResolutionStrategy
+    properties:
+    DefaultFileIDHelper: '%$SilverStripe\Assets\FilenameParsing\HashFileIDHelper'
+    ResolutionFileIDHelpers:
+      - '%$SilverStripe\Assets\FilenameParsing\HashFileIDHelper'
+      VersionedStage: Live
+  # Define protected resolution strategy
+  SilverStripe\Assets\FilenameParsing\FileResolutionStrategy.protected:
+    class: SilverStripe\Assets\FilenameParsing\FileIDHelperResolutionStrategy
+    properties:
+      DefaultFileIDHelper: '%$SilverStripe\Assets\FilenameParsing\HashFileIDHelper'
+      ResolutionFileIDHelpers:
+        - '%$SilverStripe\Assets\FilenameParsing\HashFileIDHelper'
+      VersionedStage: Stage
+```
+
+## Disabling old file resolution format
+
+When starting a brand new project, or after you’ve run the file migration task, all your files will be in their default locations. This gives you the option to disable alternative resolution formats. This will slightly speed up file resolution.
```

Comment:
```suggestion
When starting a brand new project, or after you’ve ran the file migration task, all your files will be in their default locations. This gives you the option to disable alternative resolution formats. This will slightly speed up file resolution.
```